client/src/components/Overview/ProductSummary.jsx

```diff
@@ -1,76 +1,38 @@
+import { useSelector } from 'react-redux';
 
-export default class ProductSummary extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      product: {},
-      reviewMetadata: {},
-      starPercentage: '',
-      count: ''
-    };
-  }
-
-  componentDidMount() {
-    const productId = 16060;
-    this.renderProductInfo(productId);
-    this.renderReviewInfo(productId);
-  }
-
-  renderProductInfo(productId) {
-    axios.get(`${AtelierApi.url}/products/${productId}`, {
-      headers: AtelierApi.headers
-    })
-      .then(response => {
-        this.setState({
-          product: response.data
-        });
-      })
-      .catch(err => console.log(err));
-  }
+const ProductSummary = () => {
+  const productInfo = useSelector((state) => state.product.productInfo);
 
-  renderReviewInfo(productId) {
-    axios.get(`${AtelierApi.url}/reviews/meta`, {
-      // eslint-disable-next-line camelcase
-      params: { product_id: productId },
-      headers: AtelierApi.headers
-    })
-      .then(response => {
-        this.setState({
-          reviewMetadata: response.data
-        });
-      })
-      .then(() => {
-        let ratings = 0;
-        let count = 0;
-        const eachRating = this.state.reviewMetadata.ratings;
-        for (var key in eachRating) {
-          count += Number(eachRating[key]);
-          ratings += (Number(key) * Number(eachRating[key]));
-        }
-        let starPercentage = (ratings / count) / 5 * 100;
-        // console.log(starPercentage);
-        if ((starPercentage % 5) < 2.5) {
-          starPercentage -= (starPercentage % 5);
-        } else {
-          starPercentage += (5 - (starPercentage % 5));
-        }
-
-        this.setState({
-          count: count,
-          starPercentage: starPercentage
-        }, () => {
-          // console.log(this.state.starPercentage);
-          document.querySelector('#body-overview-star .stars-inner').style.width = `${this.state.starPercentage}%`;
-        });
-
-      })
-      .catch(err => console.log(err));
-  }
+  const styleInfo = useSelector(state => {
+    if (state.product.styleInfo.results === undefined) {
+      return undefined;
+    }
+    return {
+      price: state.product.styleInfo.results[state.style]
+    };
+  });
 
-  render() {
+  console.log(styleInfo);
+  if (styleInfo && !styleInfo.price.sale_price) {
+    return (
+      <div id="body-overview-productsummary">
+        <div id="body-overview-starrating">
+          <div id="body-overview-star">
+            <div className="stars-outer">
+              <div className="stars-inner"></div>
+            </div>
+          </div>
+          <div id="body-overview-count">Read all insert number reviews</div>
+        </div>
+        <div id="body-overview-category">{productInfo.category}</div>
+        <div id="body-overview-name">{productInfo.name}</div>
+        <div id="body-overview-price">${styleInfo.price.original_price} USD</div>
```

Comment:
It would be better to have only one block of code here instead of two nearly identical blocks. The only thing that is changing is the price info based on whether the .sale_price is equal to null or not. You could use a helper function like:

```javascript
const priceHelper = (price) =>  {
if (price.sale_price === null) {
  return (<div id="body-overview-price">${price.original_price} USD</div>);
} else {
  return (
    <div id="body-overview-oldprice">${price.original_price} USD</div>
    <div id="body-overview-saleprice">${price.sale_price} USD</div>
  );
} 
```

```suggestion
        {priceHelper(price)}
```

And if you do that you don't  need the if statements, you can have just one block of code that you return.